miniscope_io/models/stream.py

```diff
@@ -68,23 +72,33 @@ class StreamDaqConfig(MiniscopeConfig, YAMLMixin):
         If `LSB`, then the incoming bitstream is in the format that each 32-bit words are bit-wise reversed on its own.
         Furthermore, the order of 32-bit words in the pixel data within the buffer is reversed (but the order of words in the header is preserved).
         Note that this format does not correspond to the usual LSB-first convention and the parameter name is chosen for the lack of better words.
+    save_video : bool, optional
+        Whether the video gets saved as an .avi file, by default False.
+    save_video_name : str, optional
+        If video is saved, , by default False.
+    show_video : bool, optional
+        Whether the video is showed in "real-time", by default True.
 
     ..todo::
         Takuya - double-check the definitions around blocks and buffers in the firmware and add description.
     """
+
+    mode: Literal["DAQ", "RAW_RECORD", "RAW_REPLAY"] = 'DAQ'
```

Comment:
ok my last question is about this parameter - i get the reasoning for this, do we think this is still useful after using the mock? the `REPLAY` is no longer used, and what do you think about replacing the `RAW_RECORD` functionality by having a flag in the `capture` method like `raw: Optional[Path] = None` to enable recording the raw binary stream? 

i think we also probably want to add some controls for capturing only a set number of frames (or else make an iterator where we can do something like `for frame in daq.iter(15)` etc. as you have done when capturing for the record mode, but maybe save for another pr?

Reply:
Making this a flag of the capture method makes sense. I first did mode because we'll want stuff like bit error check mode, etc. that users will use. However, I guess adding flags like `output: Literal["video", "raw", "BER"]` will be better once we have some cli commands that internally configure this. (Either work but I can't imagine a scenario that needs both video and RAW, so we might not need an independent flag for this.)

An iterator makes sense because I was manually changing read length, which doesn't make sense as a workflow. I guess we can do this in close future PR.

Reply:
ok did that here: https://github.com/Aharoni-Lab/miniscope-io/pull/26/commits/60221fa384bdedd94be79a752fe4e5030d5c7292

i think we'll probably want to do a more general overhaul of the notion of "sinks" or capture methods - depending on the use case, someone might want to iterate and grab any frames as fast as they can, otherwise they might want to pull from a queue to be sure they're not missing any, direct write to a video file in noninteractive mode, and so on, so i think we can save that for later and call this good for now :)